File: hw/calle/u2/hw25MongoDBConnection/ContactsBook/src/main/java/ec/edu/espe/contactsbook/controller/ContactController.java
package ec.edu.espe.contactsbook.controller;

import ec.edu.espe.contactsbook.model.Contact;
import com.mongodb.client.MongoCollection;
import com.mongodb.client.MongoDatabase;
import org.bson.Document;

public class ContactController {

    private final MongoCollection<Document> contactCollection;
    private static final String COLLECTION_NAME = "Contacts"; 
    
    public ContactController() {
        MongoDatabase db = MongoConnectionManager.getDatabase();
        if (db != null) {
            this.contactCollection = db.getCollection(COLLECTION_NAME);
        } else {
            this.contactCollection = null;
        }
    }

    
    public boolean create(Contact contact) {
        if (contactCollection == null) {
            System.err.println("Collection not available. Cannot save contact.");
            return false;
        }

        try {
            Document document = new Document()
                    .append("id", contact.getId())
                    .append("firstName", contact.getFirstName())
                    .append("lastName", contact.getLastName())
                    .append("age", contact.getAge())
                    .append("birthDate", contact.getBirthDate()) // Campo añadido
                    .append("typeOfContact", contact.getTypeOfContact())
                    .append("sex", contact.getSex())
                    .append("hobbies", contact.getHobbies())
                    .append("comments", contact.getComments());

            contactCollection.insertOne(document);
            System.out.println("Contact with ID " + contact.getId() + " successfully saved to MongoDB.");
            return true;
        } catch (Exception e) {
            System.err.println("Error inserting contact into MongoDB: " + e.getMessage());
            return false;
        }
    }
    
}

File: hw/rojas/u2/hw25MongoDBConnection/Python/ContactController.py
from Contact import ContactModel
from ValidationUtils import ValidationUtils

class ContactController:
    def __init__(self):
        self.model = ContactModel()

    def getAge(self, birthDate):
        return ValidationUtils.calculateAge(birthDate)

    def addContact(self, firstName, lastName, birthDate, age, cType, sex, hobbies, comments):
        isValid, message = ValidationUtils.validateNames(firstName, lastName)
        if not isValid:
            return False, message

        document = {
            "firstName": firstName,
            "lastName": lastName,
            "birthDate": str(birthDate),
            "age": age,
            "type": cType,
            "sex": sex,
            "hobbies": hobbies,
            "comments": comments
        }

        try:
            if self.model.save(document):
                return True, "Contact saved successfully!"
            else:
                return False, "Error connecting to Database"
        except Exception as e:
            return False, f"Error saving contact: {e}"